### DISCUSSION TITLE
intersection observer 를 useInview 형식으로 구현했는데 어떤식으로 사용하는게 좋을까요?

### DISCUSSION THREAD

**Discussion by KIMSEUNGGYU** [wanted-internship-11th-team10/pre-onboarding-11th-3-10#3] [Q&A]
## 궁금한점
infinite scroll 을 위해 맨 밑으로 이동했는지를 알아야하는데 Intersection Observer 를 통해 구현하였고 이를 재 사용하기 위해 커스텀 훅으로 추출했습니다. 
근데 사용하는 방식을 어떻게 해야하는지 궁금합니다.!

- [react-intersection-observer 참고 링크](https://www.npmjs.com/package/react-intersection-observer)

### 사용 패턴 1
- 기존 `react-intersection-observer` 패턴 방식
- 컴포넌트 내부에 useEffect 로 제어해야함 
```ts
//  IssuePage.tsx

const { ref: inViewRef, inView } = useInView<HTMLDivElement>();

useEffect(() => {
   if (!inView || isLoading) return;
  onNextFetch();
}, [inView, isLoading]);
```

### 사용 패턴 2
- callback 함수로 전달하여 제어하는 방식
```ts
const { ref: inViewRef, inView } = useInView<HTMLDivElement>({
    onInView: () => {
      if (!inView || isLoading || error) return;
      onNextFetch();
    },
  });
```

패턴1은 사용하는 곳에서 제어하는 것이고? 생명주기 관리 / 패턴2는 기능만 전달하면 내부에서 알아서 처리되는 패턴 인데 어느식으로 관리하는게 좋을지 의견이 궁금합니다.

**Comment by iinnu** [comment]
저는 콜백 함수로 전달하는 패턴이 더 나은 것 같아요.

- 옵저버 객체 만들 때 `new IntersectionObserver(callback, options)`에서 콜백을 전달
- 로직 분리 및 재사용성을 위해 커스텀 훅으로 뺌

이 2가지를 생각하면 useEffect에서 실행하는 것보다는 훅에 콜백을 전달해서 실행을 위임하는게 더 적합하지 않을까 생각합니다.

**Comment by yurjune** [comment]
저도 데이터 페칭하는 콜백함수를 인피니트 훅에 넘겼습니다.
